hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -164,13 +170,19 @@ def execute_as!(user, command, do_say = true, do_log = true, extra_env = {})
     execute!(runuser_command, do_say, do_log, extra_env)
   end
 
+  def execute_preformatted_as(user, command, do_say = true, do_log = true, extra_env = {})
+    runuser_command = ['runuser', '-l', user, '-c', command]
+    execute_command(runuser_command, do_say, do_log, extra_env)
+  end
+
   def execute(command, do_say, do_log, extra_env = {})
     _stdout_stderr, status = execute_command(command, do_say, do_log, extra_env)
     status
   end
 
   def execute_command(command, do_say, do_log, extra_env = {})
-    log_and_say(:debug, "Executing: #{command}", do_say, do_log)
+    display_command = command.is_a?(Array) ? command.join(' ') : command
```

Comment:
Should we use `shelljoin` to properly escape it?

I prefer the more explicit call:
```suggestion
    display_command = command.is_a?(Array) ? Shellwords.join(command) : command
```

But this technically also works:
```suggestion
    display_command = command.is_a?(Array) ? command.shelljoin : command
```

Be sure that `shellwords` is required somewhere

Reply:
```
2024-12-03 13:47:28 [DEBUG ] [pre] Starting services postgresql
2024-12-03 13:47:28 [DEBUG ] [root] Executing: runuser -l postgres -c psql\ -t\ -c\ \"SELECT\ 1\ FROM\ pg_database\ WHERE\ datname\ \=\ \'foreman\'\;\"
2024-12-03 13:47:28 [DEBUG ] [root]         1
2024-12-03 13:47:28 [DEBUG ] [root] 
2024-12-03 13:47:28 [DEBUG ] [pre] Updating ownership of the evr extension
2024-12-03 13:47:28 [DEBUG ] [root] Executing: runuser -l postgres -c psql\ -d\ foreman\ -c\ \"UPDATE\ pg_extension\ SET\ extowner\ \=\ \(SELECT\ oid\ FROM\ pg_authid\ WHERE\ rolname\=\'foreman\'\)\ WHERE\ extname\=\'evr\'\"
2024-12-03 13:47:28 [DEBUG ] [root] UPDATE 1
2024-12-03 13:47:28 [DEBUG ] [pre] Hook /usr/share/foreman-installer/hooks/pre/35-change-evr-extension-ownership.rb returned nil
```
Confirmed that both the shell formatted strings can be run directly on the terminal although the number of `\` is making it a little hard to read. 🤷🏼